cdm/core/src/main/resources/META-INF/services/ucar.nc2.internal.dataset.CoordSystemBuilderFactory

```diff
@@ -0,0 +1,3 @@
+ucar.nc2.internal.dataset.conv.CoardsConventions$Factory
```

Comment:
Are we missing a few (e.g. `CF1Convention`)?

Also, it feels a little odd to me to reference something under `ucar.nc2.internal` for the service provider configuration file. Is the internal one temporary until 6, at which point we use as a replacement for the current `ucar.nc2.dataset.CoordSystemBuilderFactory`?

Reply:
We're missing a lot, namely all the other conventions.
My intention is that the convention parsers are not public api.

Reply:
Im thinking most conventions will be Service Providers; the ones we explicitly register in CoordSystemFactory  are the ones we want to prioritize.

Reply:
So when using the Service Provider, a user who implements a convention will need to create a file in their jar called `META-INF/services/ucar.nc2.internal.dataset.CoordSystemBuilderFactory`? That filename (`ucar.nc2.internal.dataset.CoordSystemBuilderFactory`) is the part I'm talking about exposing "internal".

Reply:
Ok I see what youre saying. 

Yes maybe ucar.nc2.internal.dataset.CoordSystemBuilderFactory should be ucar.nc2.dataset.CoordSystemBuilderFactory.

If you agree Ill move it.

Reply:
That works for me. The only other thing I could see is putting it under `ucar.nc2.dataset.spi` if there is a chance of having other Service Providers in for `dataset`, but I think we're good either way, really. Thanks @JohnLCaron!